packages/@aws-cdk/aws-codepipeline-actions/test/cloudformation/cloudformation-pipeline-actions.test.ts

```diff
@@ -711,6 +711,571 @@ nodeunitShim({
+                  'Parameters': 'SourceArtifact::parameters.json',
+                  'Capabilities': 'CAPABILITY_NAMED_IAM',
+                  'DeploymentTargets': '11111111111,22222222222',
+                  'FailureTolerancePercentage': 50,
+                  'MaxConcurrentPercentage': 25,
+                  'Regions': 'us-east-1,us-west-1,ca-central-1',
+                },
+                'InputArtifacts': [{ 'Name': 'SourceArtifact' }],
+                'Name': 'StackSetUpdate',
+              },
+            ],
+          },
+        ],
+      }));
+
+      test.done();
+    },
+
+    'creates correct pipeline resource with parameter list'(test: Test) {
+      // GIVEN
+      const stack = new TestFixture();
+
+      const adminRole = new Role(stack, 'ChangeSetRole', {
+        assumedBy: new ServicePrincipal('cloudformation.amazonaws.com'),
+      });
+
+      stack.deployStage.addAction(new cpactions.CloudFormationStackSetAction({
+        actionName: 'StackSetUpdate',
+        description: 'desc',
+        stackSetName: 'MyStack',
+        templatePath: stack.sourceOutput.atPath('template.yaml'),
+        parameters: [
+          {
+            parameterKey: 'key0',
+            parameterValue: 'val0',
+          },
+          {
+            parameterKey: 'key1',
+            parameterValue: 'val1',
+          },
+          {
+            parameterKey: 'key2',
+            usePreviousValue: true,
+          },
+        ],
+        administrationRole: adminRole,
+        cfnCapabilities: [
+          cdk.CfnCapabilities.NAMED_IAM,
+        ],
+        executionRoleName: 'Exec',
+        deploymentTargetsPath: stack.sourceOutput.atPath('accounts.json'),
+        regions: ['us-east-1', 'us-west-1', 'ca-central-1'],
+        failureTolerancePercentage: 50,
+        maxConcurrentPercentage: 25,
+      }));
+
+      expect(stack).to(haveResourceLike('AWS::CodePipeline::Pipeline', {
+        'Stages': [
+          { 'Name': 'Source' /* don't care about the rest */ },
+          {
+            'Name': 'Deploy',
+            'Actions': [
+              {
+                'Configuration': {
+                  'StackSetName': 'MyStack',
+                  'Description': 'desc',
+                  'TemplatePath': 'SourceArtifact::template.yaml',
+                  'Parameters': 'ParameterKey=key0,ParameterValue=val0 ParameterKey=key1,ParameterValue=val1 ParameterKey=key2,UsePreviousValue=true',
+                  'Capabilities': 'CAPABILITY_NAMED_IAM',
+                  'DeploymentTargets': 'SourceArtifact::accounts.json',
+                  'FailureTolerancePercentage': 50,
+                  'MaxConcurrentPercentage': 25,
+                  'Regions': 'us-east-1,us-west-1,ca-central-1',
+                },
+                'InputArtifacts': [{ 'Name': 'SourceArtifact' }],
+                'Name': 'StackSetUpdate',
+              },
+            ],
+          },
+        ],
+      }));
+
+      test.done();
+    },
+
+  },
+
```

Comment:
Let's maybe create a new file for these tests? This file is gigantic as it is, no reason to make it even bigger I don't think...